### PR TITLE
ruby, python, node, c#: make tests faster

### PR DESCRIPTION
**Pull request by pongad**
This commit continues from #1476 and #1514.
Please see the former for more info on the speed up itself.

This reduces test execution time on my machine from ~38s
to ~27s.

### CODE REVIEW

**Comment by pongad** [comment]
I tested this change by slightly modifying a few snippets and making sure tests fail at appropriate points.

Could language owners skim the concatenated baseline files to make sure it looks sane? I apologize for the large diffs :(

**Comment by geigerj** [comment]
I think that makes sense. @lukesneeringer any objection?

File: src/main/java/com/google/api/codegen/gapic/MainGapicProviderFactory.java [hunk 1]
```diff
@@ -72,8 +72,8 @@
   public static final String NODEJS = "nodejs";
   public static final String NODEJS_DOC = "nodejs_doc";
   public static final String PHP = "php";
-  public static final String PYTHON = "python";
-  public static final String PYTHON_DOC = "python_doc";
+  public static final String PYTHON = "py";
+  public static final String PYTHON_DOC = "py_doc";
```
Comment (lukesneeringer) [inline src/main/java/com/google/api/codegen/gapic/MainGapicProviderFactory.java]: Does PYTHON_DOC even need to exist anymore?

Comment (geigerj) [reply src/main/java/com/google/api/codegen/gapic/MainGapicProviderFactory.java]: We should probably kill this, since it is replaced by your plugin.

Comment (pongad) [reply src/main/java/com/google/api/codegen/gapic/MainGapicProviderFactory.java]: Done

**Comment by pongad** [comment]
@lukesneeringer It is unfortunate but intentional. The lines are moved into one baseline file, not deleted. I unfortunately don't see a way to avoid this :(.

**Comment by pongad** [comment]
Ping @lukesneeringer are you OK with this change?

**Comment by pongad** [comment]
@chrisdunelm @geigerj I just got LGTM from Luke and resolved conflict. Could you please take another pass?

**Comment by chrisdunelm** [comment]
Still LGTM )

**Pull request merged**